server/tserver/src/main/java/org/apache/accumulo/tserver/ScanServer.java

```diff
@@ -444,16 +444,22 @@ class ScanReservation implements AutoCloseable {
     private final Collection<StoredTabletFile> files;
     private final long myReservationId;
     private final Map<KeyExtent,TabletMetadata> tabletsMetadata;
+    private final Map<TKeyExtent,List<TRange>> failures;
 
-    ScanReservation(Map<KeyExtent,TabletMetadata> tabletsMetadata, long myReservationId) {
+    /* This constructor is called when starting a scan */
+    ScanReservation(Map<KeyExtent,TabletMetadata> tabletsMetadata, long myReservationId,
+        Map<TKeyExtent,List<TRange>> failures) {
       this.tabletsMetadata = tabletsMetadata;
+      this.failures = failures;
       this.files = tabletsMetadata.values().stream().flatMap(tm -> tm.getFiles().stream())
           .collect(Collectors.toUnmodifiableSet());
       this.myReservationId = myReservationId;
     }
 
+    /* This constructor is called when continuing a scan */
```

Comment:
It seems helpful to have these comments about when each constructor is used. I wonder if creating new methods (maybe `startScan` or `continueScan`) which would use the respective constructor would be helpful or not. Might not make much of a difference, just an observation

Reply:
Do you mean create static methods (e.g. ScanReservation.startScan() and ScanReservation.continueScan()), that call a private constructor and return the ScanReservation object? I'm not sure we need that level of indirection, IMO the comments are enough.